R/zchunk_LA131.enduse.R

```diff
@@ -34,59 +36,186 @@ module_energy_LA131.enduse_DISABLED <- function(command, ...) {
+    Unoil_Refin_elect %>%
+      left_join_error_no_match(L126.out_EJ_R_electd_F_Yh, by = c("GCAM_region_ID", "fuel", "year")) %>%
+      mutate(value = value.y - value.x) %>%
+      select(-sector, -value.x, -value.y) ->
+      Enduse_elect
+
+    # Subset the end use sectors and aggregate by fuel
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      filter(sector %in% enduse_sector_aggregation$sector) %>%
+      filter(fuel == "electricity") %>%
+      group_by(GCAM_region_ID, fuel, year) %>%
+      summarise(value = sum(value)) ->
+      Enduse_elect_unscaled
+
+    # Calculate the scalers required to balance electricity within each region
+    Enduse_elect %>%
+      left_join_error_no_match(Enduse_elect_unscaled, by = c("GCAM_region_ID", "fuel", "year")) %>%
+      mutate(value = value.x / value.y) %>%
+      ungroup() %>%
+      select(GCAM_region_ID, year, value) ->
+      Enduse_elect_scaler
+
+    # Multiply the electricity scalers by the original estimates of electricity consumption by end use sectors
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      filter(sector %in% enduse_sector_aggregation$sector) %>%
+      filter(fuel == "electricity") %>%
+      left_join_error_no_match(Enduse_elect_scaler, by = c("GCAM_region_ID", "year")) %>%
+      mutate(value = value.x * value.y) %>%
+      select(-value.x, -value.y) ->
+      Enduse_elect_scaled
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # Replace unscaled estimates of end use sector electricity consumption in full table
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      filter(sector %in% enduse_sector_aggregation$sector) %>%
+      filter(fuel != "electricity") %>%
+      bind_rows(Enduse_elect_scaled) ->
+      Enduse_elect_scaled_heat_unscaled # still need to scale heat
+
+    # HEAT SCALING
+    # Total delivered heat = output of district heat sector + secondary (heat) output of electric sector
+    L124.out_EJ_R_heatfromelec_F_Yh %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value)) ->
+      Heatfromelect
+
+    L124.out_EJ_R_heat_F_Yh %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value)) %>%
+      left_join_error_no_match(Heatfromelect, by = c("GCAM_region_ID", "year")) %>%
+      mutate(value = value.x + value.y) %>%
+      select(-value.x, -value.y) ->
+      Enduse_heat
+
+    # Subset the end use sectors and aggregate by fuel. Only in regions where heat is modeled as a separate fuel.
+    A_regions %>%
+      filter(heat == 1) %>%
```

Comment:
You're right. I had thought that the comment in line 134 explained it, but it's not a direct explanation. I'll make it clearer.